tests/unit/store/createStore.ts

```diff
@@ -370,5 +380,80 @@ registerSuite({
 				assert.deepEqual(result, ids, 'Should have returned all deleted ids');
 			}).then(dfd.resolve);
 		}
+	},
+
+	'async storage': {
+		'async operation should not be done immediately.'(this: any) {
+			const{ dfd, store } = getStoreWithAsyncStorage(this, { put: 21 });
+
+			const start = Date.now();
+			store.add(createData()).then(dfd.callback(function(result: ItemType[]) {
+				const finish = Date.now();
+				assert.isAbove(finish - start, 19 );
+				dfd.resolve();
+			}));
+		},
+		'should complete initial add before subsequent operations'(this: any) {
+			const dfd = this.async(1000);
```

Comment:
For simpler tests like this one, returning a promise is enough. `this.async` used here is for consistency with other places where using `const dfd = this.async(1000);` is necessary. For example, we have a[ bunch of tests](https://github.com/dojo/stores/blob/master/tests/unit/store/mixins/createObservableStoreMixin.ts#L177-L195) that verifies a series of returned value over time. 

Basically this approach allows us to have finer-grained control over test flow.  Also `this.async()` allows us to override the default timeout `30000`.